### PR TITLE
Fix SILGen comments for isPlusOne and isPlusOneOrTrivial.

### PR DESCRIPTION
**Pull request by atrick**
Clarify a new API that landed earlier today.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test

**Review by gottesmm — APPROVED** [review]
Thanks! LGTM

File: lib/SILGen/ManagedValue.h [hunk 1]
```diff
@@ -219,20 +219,29 @@ class ManagedValue {
...
+  /// a cleanup regardless of whether the type is trivial.
   ///
-  /// When an SSA value does not have ownership, it can be used by a consuming
-  /// operation without destroying it. Consuming a value by address, however,
-  /// deinitializes the memory regardless of whether the value has ownership.
+  /// When an object value is trivial, it can be passed to a consuming operation
+  /// without destroying it. Consuming a value by address, however, always
+  /// deinitializes the memory regardless of its type.
+  ///
+  /// Use this before emitting an operation that "takes" this vaule or passing
```
Comment (gottesmm) [inline lib/SILGen/ManagedValue.h:230]: vaule -> value.

File: lib/SILGen/ManagedValue.h [hunk 2]
```diff
@@ -219,20 +219,29 @@ class ManagedValue {
...
   bool isPlusOne(SILGenFunction &SGF) const;
 
   /// Returns true if this managed value can be forwarded without necessarilly
   /// destroying the original.
   ///
-  /// This is true if either isPlusOne is true or the value is trivial. A
-  /// trivial value in memory can be forwarded as a +1 value without
-  /// deinitializing the memory.
+  /// This is true if either isPlusOne is true or the value is trivial. Unlike
+  /// isPlusOne(), this returns true for trivial addres values. A +1 value can
```
Comment (gottesmm) [inline lib/SILGen/ManagedValue.h:238]: addres -> address.

I would also say this returns true for a trivial address value regardless of whether or not it has a cleanup attached.

File: lib/SILGen/ManagedValue.h [hunk 3]
```diff
@@ -219,20 +219,29 @@ class ManagedValue {
...
+  /// This is true if either this value has a cleanup or if it is a trivial
+  /// object value. For address values, this returns true only if the value has
+  /// a cleanup regardless of whether the type is trivial.
   ///
-  /// When an SSA value does not have ownership, it can be used by a consuming
-  /// operation without destroying it. Consuming a value by address, however,
-  /// deinitializes the memory regardless of whether the value has ownership.
+  /// When an object value is trivial, it can be passed to a consuming operation
+  /// without destroying it. Consuming a value by address, however, always
+  /// deinitializes the memory regardless of its type.
```
Comment (gottesmm) [inline lib/SILGen/ManagedValue.h:228]: the memory regardless of its type -> the memory regardless of whether or not it is trivial.

File: lib/SILGen/ManagedValue.h [hunk 4]
```diff
@@ -219,20 +219,29 @@ class ManagedValue {
...
   ///
-  /// This is true if either isPlusOne is true or the value is trivial. A
-  /// trivial value in memory can be forwarded as a +1 value without
-  /// deinitializing the memory.
+  /// This is true if either isPlusOne is true or the value is trivial. Unlike
+  /// isPlusOne(), this returns true for trivial addres values. A +1 value can
+  /// be created from a trivial value without consuming the original.
+  ///
+  /// Use this when storing this value into a new location simply by forwarding
+  /// the cleanup without destroying the original value. If it's necessary to
```
Comment (gottesmm) [inline lib/SILGen/ManagedValue.h:242]: Nice!

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Comment by atrick** [comment]
@swift-ci smoke test and merge

**Pull request merged**